src/Host.cpp

```diff
@@ -3809,22 +3809,36 @@ std::optional<QString> Host::windowType(const QString& name) const
         return {QStringLiteral("label")};
     }
 
-    if (mpConsole->mSubConsoleMap.contains(name)) {
-        auto userwindow = mpConsole->mDockWidgetMap.value(name);
-        if (userwindow) {
+    if (name == QLatin1String("main")) {
+        return {QLatin1String("main")};
+    }
+
+    auto pWindow = mpConsole->mSubConsoleMap.value(name);
+    if (pWindow) {
+        switch (pWindow->getType()) {
+        case TConsole::UserWindow:
             return {QStringLiteral("userwindow")};
+        case TConsole::Buffer:
+            return {QStringLiteral("buffer")};
+        case TConsole::SubConsole:
+            return {QStringLiteral("miniconsole")};
+        case TConsole::UnknownType:
```

Comment:
🙄 What a ridiculous warning - the whole purpose of using the "identical" branch code in this case is because otherwise the same bot is going to moan about something like "*this code will fall-through to the next case*" ...

Reply:
This is running `clang-tidy` which you can check with locally by the way